spec/datadog/tracing/configuration/settings_spec.rb

```diff
@@ -503,6 +503,45 @@ def propagation_style_inject
       end
     end
 
+    describe '#native_span_events' do
+      subject(:native_span_events) { settings.tracing.native_span_events }
+
+      context "when #{Datadog::Tracing::Configuration::Ext::ENV_NATIVE_SPAN_EVENTS}" do
+        around do |example|
+          ClimateControl.modify(Datadog::Tracing::Configuration::Ext::ENV_NATIVE_SPAN_EVENTS => environment) do
+            example.run
+          end
+        end
+
+        context 'is not defined' do
+          let(:environment) { nil }
+
+          it { is_expected.to be nil }
+        end
+
+        context 'is set to true' do
+          let(:environment) { 'true' }
+
+          it { is_expected.to be true }
+        end
+
+        context 'is set to false' do
+          let(:environment) { 'false' }
+
+          it { is_expected.to be false }
+        end
+      end
```

Comment:
Minor: There's a helper in the `core/configuration/settings_spec.rb` that may be useful to use here. It allows this to be replaced with

```ruby
it_behaves_like 'a binary setting with', env_variable: 'DD_TRACE_NATIVE_SPAN_EVENTS', default: nil
```

Reply:
Wow, I didn't know about it! I'll refactor it!